pkg/driver/driver.go

```diff
@@ -48,7 +48,7 @@ type Driver struct {
 func NewDriver(endpoint string) (*Driver, error) {
 	klog.Infof("Driver: %v Version: %v", DriverName, driverVersion)
 
-	cloud, err := cloud.NewCloud()
+	cloud, err := cloud.NewCloud(nil)
```

Comment:
Could you create the metadata service and pass it in here? Im trying to avoid passing in nil.

Reply:
The problem with that is that its going to leak AWS implementation details into this driver code. If we are ok with that I can make the change.
Another problem is that `EC2Metadata` returned by `svc := ec2metadata.New(sess)` is also required in `&ec2rolecreds.EC2RoleProvider{Client: svc},`, passing in both also doesn't seem right. 
Maybe we can have a new function, something like below that is used in the e2e tests, and change back the original function `func NewCloud()  (Cloud, error)`:
```
func NewCloudWithMetadata(metadata MetadataService) (Cloud, error)
```

Reply:
`NewCloudWithMetadata` sounds a better choice for me. 

I feel the way we are handling metadata is a bit messy. Will need to refactor later to improve this.

Reply:
Sounds good, I'll make the change and test the code locally and push it up soon. But you are right, the metadata code needs some refactoring the interaction with `MetadataService` and `EC2Metadata` is confusing.